### PR TITLE
Configuration in one file

### PR DESCRIPTION
**Pull request by xamanu**
**tldr;** This PR introduces a new way of handling configuration a bit more easily and consistent in one configuration file - [`tasking-manager.env`](https://github.com/hotosm/tasking-manager/blob/modify/Configuration/example.env).

--------

* All configuration can live now in a new `tasking-manager.env`. Copy over the `example.env`, add or adjust the variables you need.
* The PR is largely backwards compatible (you can still use the environment variables as before). However, **there is one relevant breaking change** introduced here: 
   - `TM_ENV` has become deprecated, instead administrators have to specify the `TM_APP_BASE_URL`.

Some additional notes:
* This PR is based on #1416 and #1417.
* Please have a look at the edited [README.md](https://github.com/hotosm/tasking-manager/blob/modify/Configuration/README.md) for more information on how a setup will work with this PR.
* Make sure you use the repository's `server/config.py` - there shall be no reason to alter this anymore.

---------

### Testing instructions

There are two ways for testing this. Both have to work, in order to get this PR in:

#### 1. Bare bone setup (development setup)

* Clone the Tasking Manager in a new location
* Use a different branch: `git checkout modify/Configuration`
* Copy `example.env` to `tasking-manager.env`
* Adjust variables in `tasking-manager.env`
* Set `TM_LOG_DIR=logs` in the `tasking-manager.env`
* Run the front-end building and python environment preparation
* Run the server and see whether it works

#### 2. Docker-Compose setup

* Install a recent version of [Docker](https://docs.docker.com/get-started/) and [Docker Compose](https://docs.docker.com/compose/).
* Check out the code: `git clone https://github.com/hotosm/tasking-manager.git && cd tasking-manager`
* Use a different branch: `git checkout modify/Configuration`
* Copy the configuration file `cp env-example tasking-manager.env`
* Add the consumer key and secret in the `tasking-manager.env` file.
* Run the system: `sudo docker-compose build && sudo docker-compose up`
* Open another terminal and build the front-end: `sudo docker exec -it tasking-manager_app_1 /bin/bash -c 'cd client && npm install && gulp build'`
* Test whether logging in works under http://127.0.0.1:5000

### CODE REVIEW

**Comment by xamanu** [comment]
Recently we had some issues onboarding interested programmers (#1388 and #1391). This setup is much easier to troubleshoot and has been confirmed to work for them.

File: example.env [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Tasking Manager configuration file
+#
+
+
+#
+# Rename to `.env` and adjust to make it work!
```
[the remark this answers is not in the document]
Comment (xamanu) [reply example.env]: Fixed. Thanks for spotting.

File: example.env [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# TM_SMTP_PORT=25
+# TM_SMTP_USER=
+# TM_SMTP_PASSWORD=
+
+# Logging settings  (optional)
+# (e.g. ERROR, DEBUG, etc.)
+# If not specified DEBUG is default. ERROR is a good value for a live site.
+#
+# TM_LOG_LEVEL=DEBUG
+# TM_LOG_DIR=/var/log/tasking-manager.logs
```
[the remark this answers is not in the document]
Comment (xamanu) [reply example.env]: Yes, I think this is ideally `# TM_LOG_DIR=logs`. Changed both the `config.py` and the `example.env` accordingly.

File: README.md [hunk 3]
```diff
@@ -150,7 +142,7 @@ The project includes a suite of Unit and Integration tests that you should run a
 python -m unittest discover tests/server
 ```
 
-## Database
+### Database
```
[the remark this answers is not in the document]
Comment (xamanu) [reply README.md]: Yes, indeed. I will update as soon as this is becoming released.

**Comment by xamanu** [comment]
Thanks for the reviews, @ramyaragupathy and @dakotabenjamin. Merging this into `develop`.
